src/applet/profile/download.c

```diff
@@ -37,6 +40,15 @@ char *strsep(char **stringp, const char *__delim) {
 }
 #endif
 
+static bool is_strict_matching_id(const char *token) {
+    const size_t n = strlen(token);
+    for (int i = 0; i < n; i++) {
+        if (isalnum(token[i]) || token[i] == '-') continue;
```

Comment:
Don't use `isalnum()`, it will contain other characters when the locale is not "C.UTF-8".

Reply:
Or add `setlocale(LC_ALL, "C.UTF-8");` to https://github.com/estkme-group/lpac/blob/f043c01a0028ee4eb0ba374dd771cb90b4770427/src/main.c#L166, and then you can use not `*_l` version everywhere except libeuicc.